generic3g/ComponentSpecParser/parse_var_specs.F90

```diff
@@ -102,10 +109,12 @@ subroutine parse_state_specs(var_specs, hconfig, state_intent, rc)
                  default_value=default_value, &
                  vertical_dim_spec=vertical_dim_spec, &
                  ungridded_dims=ungridded_dims, &
-                 dependencies=dependencies &
+                 dependencies=dependencies, &
+                 accumulation_type=accumulation_type &
                  )
             if (allocated(units)) deallocate(units)
             if (allocated(standard_name)) deallocate(standard_name)
+            if (allocated(accumulation_type)) deallocate(accumulation_type)
```

Comment:
Just a reminder to myself.  This "antipattern" (needing to explicitly deallocate) strongly suggests that the innards of  the loop should be refactored into a function where the deallocation would happen naturally on exit of the function.

Reply:
Makes sense.